infrastructure/tasks/10-functions.ts

```diff
@@ -57,15 +57,17 @@ export const run = async () => {
   );
   const storageAccountKeys = await storageClient.storageAccounts.listKeys(
     config.azurerm_resource_group,
-    config.azurerm_storage_account
+    config.azurerm_functionapp_storage_account
   );
   if (!storageAccountKeys || !storageAccountKeys.keys) {
-    throw new Error("storageAccountKeys not found");
+    throw new Error("Functions storage account keys not found");
   }
   // We finally got the storage account keys so we can build the connection string
   // @see StorageAccountListKeysResult
-  const storageConnectionString = `DefaultEndpointsProtocol=https;AccountName=${config.azurerm_storage_account};AccountKey=${storageAccountKeys
-    .keys[0].value}`;
+  const storageConnectionString =
+    `DefaultEndpointsProtocol=https;AccountName=` +
+    `${config.azurerm_functionapp_storage_account};AccountKey=${storageAccountKeys
```

Comment:
for clarity, I would suggest to assign `storageAccountKeys.keys[0].value` to a variable with a meaningful name and use that here